### ISSUE TITLE
cluster_between_groups and col_split_order: allow custom col_split_order after clustering the groups specified with col_split

### ISSUE THREAD

**Issue by stephanmg** [DingWB/PyComplexHeatmap#57]
**Is your feature request related to a problem? Please describe.**
Desire to use col_split and col_split_order to specify my column splits and the order in combination with clustering between groups.

**Describe the solution you'd like**
I would like to control the col_split_order after cluster_between_groups has been used to do the mean clustering.

**Describe alternatives you've considered**
None, but I could use cluster_between_groups and use a vector graphics program for instance to reorder the groups which would be a bit messy. 

**Additional context**

**Comment by DingWB** [comment]
If you wanted to use custom split_order, then you should not use cluster_between_groups. For cluster_between_groups, the split order would be inferred automatically.

**Comment by stephanmg** [comment]
@DingWB yes I am aware, can one combine both?

**Comment by DingWB** [comment]
Do you have any suggestions about how to combine both?
What if the user-given split order was inconsistent with the clustering tree?

**Comment by stephanmg** [comment]
> Do you have any suggestions about how to combine both? What if the user-given split order was inconsistent with the clustering tree?

Okay, you are right, I assume here that I have already a row clustering for my matrix. Now, what I really want is, I want to do a row split to highlight groups by introducing whitespace (depending on cutting the row dendrogram at a certain threshold). 

I can do this by specifying a row_split, but then I loose the full row dendrogram. If I use the cluster_between_groups I loose the controll of the order of the groups in the rows. 

Basically want to do a row_split at certain indices but keep the full dendrogram.

**Comment by stephanmg** [comment]
No immediate suggestion, just would be nice to be able to do this automatically. 
Thanks.

**Comment by DingWB** [comment]
I will try to implement the full dendrogram when `row_split` or `col_split` is specified, maybe this weekend.

**Comment by stephanmg** [comment]
Cool! Highly appreciated.

**Comment by DingWB** [comment]
It has already been implemented in the latest version (1.6.3), please check it out.
Please test it and give me feedback if there is any bug or suggestions. @stephanmg

**Comment by stephanmg** [comment]
@DingWB thanks, how would I use it. specify cluster_between_groups and the col_split_order simulatenously?

**Comment by DingWB** [comment]
It doesn't make sense to show the full dendrogram and specify custom split orders simultaneously.
In the latest version, If you specify `row_split` or `col_split`, the full dendrogram will show automatically.
